proposals/dynamic-network-version-detection.md

```diff
@@ -0,0 +1,306 @@
+# Dynamic Network Version Detection and Feature Adaptation
+
+**Issue**: #36 - Enable SDK to Detect Network Version and Adjust Logic Dynamically  
+**Author**: Solution Proposal  
+**Date**: November 26, 2025
+
+## Problem Statement
+
+The Hiero SDK currently faces challenges with feature rollout timing and backward compatibility:
+
+1. **Delayed Feature Support**: Features like HIP-1021 and HIP-1086 had to be delayed until mainnet deployment, preventing early access on testnet/previewnet
+2. **Backward Compatibility Issues**: Newer SDK versions can break compatibility with older consensus node versions
+3. **Manual Version Management**: Developers must manually track which network versions support which features
+
+## Current State Analysis
+
+### Available Version Information
+
+Based on the protobuf definitions, consensus nodes expose version information through:
+
+- **`NetworkGetVersionInfoQuery`**: Returns semantic version information
+- **`SemanticVersion`**: Contains major, minor, patch, and pre-release components
+- **Two version types**:
+  - `hapiProtoVersion`: Hedera API (HAPI) protobuf message version
+  - `hederaServicesVersion`: Hedera Services software version
+
+### Version Format
+```protobuf
+message SemanticVersion {
+    int32 major = 1;    // Always 0 for Hedera (API may change)
+    int32 minor = 2;    // Increments each release
+    int32 patch = 3;    // Typical patch versioning
+    string pre = 4;     // Pre-release identifier
+}
+```
+
+## Proposed Solution Architecture
+
+### 1. Version Detection Service
+
+Create a centralized `NetworkVersionService` that:
+
+```java
+public interface NetworkVersionService {
```

Comment:
Can you please add a short documentation to all methods. Best would be to not specify it in Java but already use our guideline for defining API changes and new APIs in SDKs: https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/api-guideline.md
By doing so you can define it in a language independent way. We assume that this will become a must in future and it would be great to see it used in proposals today to check if important specifications are missing in the guideline.